### PR TITLE
Fix apping values that comprise only of an anchor

### PR DESCRIPTION
**Pull request by kbairak**
Problem and/or solution
-----------------------

The YaML parser is made to ignore anchors since they are irrelevant to localization and they are vulnerable to "billion laughs" attacks. So, when it encounters a:

```yaml
key: &anchor value
```

It will only consider the "value" part to be part of the extracted string. The problem is that, when the anchor **is** encountered, the
parser _expects_ there to be a value afterwards. When this is encountered:

```yaml
key: &anchor
```

according to the YAML spec, it should be equivalent to:

```yaml
key:
```

meaning the value of the key should be `null`.

```python
yaml.safe_load("key: &anchor")
# <<< {'key': None}
```

However, because the handler assumes that there is a value following the anchor, it does a `...split(' ', 1)[1]` which raises a `ValueError`.

The fix is it check against the captured YaMl node after an anchor is encountered and, if it's a null node, return it as-is.

How to test
-----------

Try uploading this file:

```yaml
key1: value1
key2: &anchor2 value2
key3: &anchor3
```

on `devel` and on this branch, which is what the brand-new unittest does anyway.

Reviewer checklist
------------------

Code:
* [ ] Change is covered by unit-tests
* [ ] Code is well documented, well styled and is following [best practices](https://tem.transifex.com)
* [ ] Performance issues have been taken under consideration
* [ ] Errors and other edge-cases are handled properly

PR:
* [ ] Problem and/or solution are well-explained
* [ ] Commits have been squashed so that each one has a clear purpose
* [ ] Commits have a proper commit message [according to TEM](https://tem.transifex.com/github-guide.html#working-on-a-feature)

### CODE REVIEW

File: openformats/formats/yaml/utils.py [hunk 1]
```diff
@@ -81,15 +81,19 @@ def compose_scalar_node(self, anchor):
...
-           anchor_value = self.stream[
-               node.start_mark.index:node.end_mark.index
-            ].split(' ', 1)[1]
-           leading_spaces = len(anchor_value) - len(anchor_value.lstrip(' '))
-
-           node.start_mark.index = node.end_mark.index - len(anchor_value) \
-               + leading_spaces
-           return node
+            node = super(TxYamlLoader, self).compose_scalar_node(anchor)
+            if node.tag == u'tag:yaml.org,2002:null':
```
Comment (kostasgian) [inline openformats/formats/yaml/utils.py:85]: I suppose this has an reasonable explanation, but since I am ignorant of what's going on under the hood, matching the node tag with this specific equation seems kind strange to me... 

So the tag is always coming as such when anchor value is null?

Comment (kostasgian) [reply openformats/formats/yaml/utils.py:85]: This is more a question to understand how this works... Tests are showing that indeed the solution works.

Comment (kbairak) [reply openformats/formats/yaml/utils.py:85]: That's a good question. TBH, I didn't check the docs to see what value to expect from `node.tag`. What I did was run the code through a debugger and inspect the value.

Comment (grass-tx) [reply openformats/formats/yaml/utils.py:85]: @kostasgian I think [this](https://yaml.org/spec/1.2/spec.html#id2805071) answers your question and also attests to the validity of @kbairak's findings.

Comment (deathbird) [reply openformats/formats/yaml/utils.py:85]: Browsing [PyYAML's docs](https://pyyaml.docsforge.com/master/documentation/#nodes) it seems a safe assumption.

**Review by deathbird — APPROVED** [review]
It works, Good job @kbairak!

File: openformats/formats/yaml/utils.py (hunk 1 above)
Comment (kostasgian) [reply openformats/formats/yaml/utils.py:85]: Thanks @grass-tx and @deathbird for the clarifications!

Comment (kostasgian) [reply openformats/formats/yaml/utils.py:85]: I am adding here a small excerpt from the yaml specification, only for future reference. 
"If none of the regular expressions matches, the scalar is resolved to `tag:yaml.org,2002:str`"

**Review by kostasgian — APPROVED** [review]
LGTM
